### PR TITLE
#7: add Tag class with predefined tags

### PR DESCRIPTION
**Pull request by hohwille**
#7: add Tag class with predefined tags including taxonomy/hierarchy add synonyms

### CODE REVIEW

**Review by jan-vcapgemini — COMMENTED** [review]
These new tags are really nice.
I've fixed some typos and added one question. Looks good to me.

**Review by aBega2000 — APPROVED** [review]
This implementation of Tags looks really nice. Jan already fixed some small typos, and from my side everything else looks correct.

File: cli/src/main/java/com/devonfw/tools/ide/tool/ide/IdeToolCommandlet.java [hunk 1]
```diff
@@ -38,10 +39,9 @@ public abstract class IdeToolCommandlet extends LocalToolCommandlet {
    * @param tags the {@link #getTags() tags} classifying the tool. Should be created via {@link Set#of(Object) Set.of}
    *        method.
    */
-  public IdeToolCommandlet(IdeContext context, String tool, Set<String> tags) {
+  public IdeToolCommandlet(IdeContext context, String tool, Set<Tag> tags) {
 
     super(context, tool, tags);
-    assert (tags.contains(TAG_IDE));
```
[the remark this answers is not in the document]
Comment (hohwille) [reply cli/src/main/java/com/devonfw/tools/ide/tool/ide/IdeToolCommandlet.java:44]: Addressed. @jan-vcapgemini can you resolve so I can merge?

**Review by jan-vcapgemini — COMMENTED** [review]
Looks fine to me.

**Pull request merged**